oauth1a.go

```diff
@@ -194,20 +162,55 @@ func (s *HmacSha1Signer) Sign(request *http.Request, clientConfig *ClientConfig,
 	oauthHeader := "OAuth " + strings.Join(headerParts, ", ")
 	request.Header["Authorization"] = []string{oauthHeader}
 
-	// This bit of fussing is because '/' is not encoded correctly
-	// by the URL package, so we encode manually.
-	values = request.URL.Query()
-	if len(values) > 0 {
-		buf = bytes.NewBufferString("")
-		for key, val := range values {
-			buf.Write([]byte("&"))
-			buf.Write([]byte(Rfc3986Escape(key)))
-			buf.Write([]byte("="))
-			buf.Write([]byte(Rfc3986Escape(val[0])))
+	request.URL.RawQuery = sortedQueryString(request.URL.Query())
+	return nil
+}
+
+// This bit of fussing is because '/' is not encoded correctly
+// by the URL package, so we encode manually.
+func sortedQueryString(values url.Values) string {
+	if len(values) == 0 {
+		return ""
+	}
+	pairs := make(sortedPairs, 0)
+	for k, vs := range values {
+		escK := Rfc3986Escape(k)
+		for _, v := range vs {
+			pairs = append(pairs, pair{escK, Rfc3986Escape(v)})
 		}
-		request.URL.RawQuery = buf.String()[1:]
 	}
-	return nil
+	sort.Sort(pairs)
+
+	buf := new(bytes.Buffer)
+	buf.WriteString(pairs[0].k)
+	buf.WriteByte('=')
+	buf.WriteString(pairs[0].v)
+
+	for _, p := range pairs[1:] {
+		buf.WriteByte('&')
+		buf.WriteString(p.k)
+		buf.WriteByte('=')
+		buf.WriteString(p.v)
+	}
+	return buf.String()
+}
+
+type pair struct{ k, v string }
+type sortedPairs []pair
+
+func (sp sortedPairs) Len() int {
+	return len(sp)
+}
+func (p sortedPairs) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+func (p sortedPairs) Less(i, j int) bool {
+	sgn := strings.Compare(p[i].k, p[j].k)
```

Comment:
strings.Compare is relatively new, in Go 1.5: https://golang.org/doc/go1.5

I'd prefer to avoid forcing an upgrade given that 1.5 came out on August 19th.  It looks like the following should work as well:
```
	if p[i].k == p[j].k {
		return p[i].v < p[j].v
	}
	return p[i].k < p[j].k
```